### DISCUSSION TITLE
Token names exported in camelcase

### DISCUSSION THREAD

**Discussion by glouhaichi** [mikaelvesavuori/figmagic#147] [Q&A] [answered]
We are experimenting with Figmagic in our project to export tokens created in our Figma files. The names in the figma files are using kebab case (ex. `$ixc-cognira-opacity-error500-005`), but after importing the tokens in the project with `npm run figmagic`, the names change to camel case (ex. `ixcCogniraOpacityError500005`).

As you can see, the case has changed and the `$` sign is gone. This presents an issue for us because we are using a naming convention where each name consists of 5 parts, and trying to decmopose the name into its constituants is impossible in the case of the example above (no way to know at which point to chop the number).

Is there a way to export the names in their original case? I scoured through the documentation but have not found a way around this. Am I missing a setting or is this not possible?

**Comment by mikaelvesavuori** [comment, marked as answer]
Hi! Correct, this is the intended behavior. We could turn this into an issue and I can see when I have time (or if someone else picks it up) to fix support, for non-recased names (which may or may not be easy to fix, because of various instances where matching happens).

When it comes to the dollar sign, I am curious if that is "correct" when it comes to JS/TS/JSON...? Would it work (in your case) without the dollar sign? (since that would imply we have to potentially support any character, which doesn't seem ideal)

**Reply by glouhaichi** [reply]
Hi, thanks for the prompt response. We would love to be able to import the names as they are without transformation, if possible, otherwise, we'd have to redesign the names of some tokens to avoid uncertainty in decomposing the name from the application side. As for the dollar sign, it was just an oddity that I had noticed, it doesn't cause us any problem, I just thought I'd mention it.

Would you suggest we create an issue for this? What would be the next steps from here if you guys believe this is worth addressing?

**Reply by mikaelvesavuori** [reply]
I can convert this (doing so now). Next steps is basically that I (or someone) starts work on the issue. You can find the overall contribution notes at https://github.com/mikaelvesavuori/figmagic/blob/main/CONTRIBUTING.md if you wish to work on this.

**Reply by mikaelvesavuori** [reply]
See: https://github.com/mikaelvesavuori/figmagic/issues/148
